--- phpBB3-init/includes/functions_user.php	2011-07-10 22:36:17.000000000 +0200
+++ phpBB3/includes/functions_user.php	2013-07-12 14:16:10.469725705 +0200
@@ -172,16 +172,16 @@
 		'user_pass_convert'	=> 0,
 		'user_email'		=> strtolower($user_row['user_email']),
 		'user_email_hash'	=> phpbb_email_hash($user_row['user_email']),
 		'group_id'			=> $user_row['group_id'],
 		'user_type'			=> $user_row['user_type'],
+		'user_timezone'         => $config['board_timezone'],
 	);
 
 	// These are the additional vars able to be specified
 	$additional_vars = array(
 		'user_permissions'	=> '',
-		'user_timezone'		=> $config['board_timezone'],
 		'user_dateformat'	=> $config['default_dateformat'],
 		'user_lang'			=> $config['default_lang'],
 		'user_style'		=> (int) $config['default_style'],
 		'user_actkey'		=> '',
 		'user_ip'			=> '',
@@ -3585,6 +3585,6 @@
 	}
 
 	return $user_data['group_id'];
 }
 
-?>
\ Pas de fin de ligne à la fin du fichier
+?>
