--- default.rb.origin	2014-09-24 10:01:19.000000000 +0200
+++ default.rb	2014-12-15 16:45:51.016214501 +0100
@@ -33,11 +33,11 @@ if File.exists?("/etc/gitlab/gitlab.rb")
 end
 node.consume_attributes(Gitlab.generate_config(node['fqdn']))
 
-if File.exists?("/var/opt/gitlab/bootstrapped")
+if File.exists?("@DATAS/bootstrapped")
 	node.set['gitlab']['bootstrap']['enable'] = false
 end
 
-directory "/var/opt/gitlab" do
+directory "@DATAS/" do
   owner "root"
   group "root"
   mode "0755"
