--- edit.html.haml	2014-12-17 13:55:19.560225135 +0100
+++ edit-patched.html.haml	2014-12-17 14:48:46.508327758 +0100
@@ -11,12 +11,16 @@
           %fieldset
             .form-group.project_name_holder
               = f.label :name, class: 'control-label' do
                 Project name
               .col-sm-10
-                = f.text_field :name, placeholder: "Example Project", class: "form-control"
-
+                - if can?(current_user, :admin_project_info, @project)
+                  = f.text_field :name, placeholder: "Example Project", class: "form-control"
+                - else
+                  = f.text_field :name, placeholder: "Example Project", class: "form-control", readonly: true
+                  %span.help-block.light
+                    Project name is read-only for NovaForge project
 
             .form-group
               = f.label :description, class: 'control-label' do
                 Project description
                 %span.light (optional)
@@ -127,28 +131,31 @@
                     data: { confirm: "Are you sure that you want to archive this project?\nAn archived project cannot be committed to." },
                     method: :post, class: "btn btn-warning"
         - else
           .nothing-here-block Only the project owner can archive a project
 
-        .panel.panel-default.panel.panel-warning
-          .panel-heading Rename repository
-          .errors-holder
-          .panel-body
-            = form_for(@project, html: { class: 'form-horizontal' }) do |f|
-              .form-group
-                = f.label :path, class: 'control-label' do
-                  %span Path
-                .col-sm-9
-                  .form-group
-                    .input-group
-                      = f.text_field :path, class: 'form-control'
-                      %span.input-group-addon .git
-                  %ul
-                    %li Be careful. Renaming a project's repository can have unintended side effects.
-                    %li You will need to update your local repositories to point to the new location.
-              .form-actions
-                = f.submit 'Rename', class: "btn btn-warning"
+        - if can?(current_user, :rename_project, @project)
+          .panel.panel-default.panel.panel-warning
+            .panel-heading Rename repository
+            .errors-holder
+            .panel-body
+              = form_for(@project, html: { class: 'form-horizontal' }) do |f|
+                .form-group
+                  = f.label :path, class: 'control-label' do
+                    %span Path
+                  .col-sm-9
+                    .form-group
+                      .input-group
+                        = f.text_field :path, class: 'form-control'
+                        %span.input-group-addon .git
+                    %ul
+                      %li Be careful. Renaming a project's repository can have unintended side effects.
+                      %li You will need to update your local repositories to point to the new location.
+                .form-actions
+                  = f.submit 'Rename', class: "btn btn-warning"
+        - else
+          .nothing-here-block Only the project owner can rename a project
 
         - if can?(current_user, :change_namespace, @project)
           .panel.panel-default.panel.panel-danger
             .panel-heading Transfer project
             .errors-holder
