--- show.html.haml	2014-12-17 14:44:14.540319055 +0100
+++ show-patched.html.haml	2014-12-17 14:51:40.876333338 +0100
@@ -17,21 +17,26 @@
   .row
     .col-md-7
       .form-group
         = f.label :name, class: "control-label"
         .col-sm-10
-          = f.text_field :name, class: "form-control", required: true
-          %span.help-block Enter your name, so people you know can recognize you.
+          - if current_user.admin?
+            = f.text_field :name, class: "form-control", required: true
+            %span.help-block Enter your name, so people you know can recognize you.
+          - else
+            = f.text_field :name, class: "form-control", required: true, readonly: true
+            %span.help-block.light
+              Name is read-only for NovaForge user
 
       .form-group
         = f.label :email, class: "control-label"
         .col-sm-10
           - if @user.ldap_user?
             = f.text_field :email, class: "form-control", required: true, readonly: true
             %span.help-block.light
               Email is read-only for LDAP user
-          - else
+          - elsif current_user.admin?
             - if @user.temp_oauth_email?
               = f.text_field :email, class: "form-control", required: true, value: nil
             - else
               = f.text_field :email, class: "form-control", required: true
             - if @user.unconfirmed_email.present?
@@ -39,10 +44,14 @@
                 Please click the link in the confirmation email before continuing, it was send to
                 %strong #{@user.unconfirmed_email}
 
             - else
               %span.help-block We also use email for avatar detection if no avatar is uploaded.
+          - else
+            = f.text_field :email, class: "form-control", required: true, readonly: true
+            %span.help-block.light
+              Email is read-only for NovaForge user
       .form-group
         = f.label :skype, class: "control-label"
         .col-sm-10= f.text_field :skype, class: "form-control"
       .form-group
         = f.label :linkedin, class: "control-label"
