--- session_store.rb	2014-12-16 17:01:03.849816943 +0100
+++ session_store-patched.rb	2014-12-16 17:00:47.313816414 +0100
@@ -1,9 +1,10 @@
 # Be sure to restart your server when you modify this file.
 
 Gitlab::Application.config.session_store(
-  :redis_store, # Using the cookie_store would enable session replay attacks.
+  :active_record_store, # NovaForge :: Use active_record_store in order to store CAS Service Ticket for the session
+  #:redis_store, # Using the cookie_store would enable session replay attacks.
   servers: Gitlab::Application.config.cache_store[1].merge(namespace: 'session:gitlab'), # re-use the Redis config from the Rails cache store
   key: '_gitlab_session',
   secure: Gitlab.config.gitlab.https,
   httponly: true,
   expire_after: 1.week,
