diff -Naur mail_tt2/automatic_bounce_management.tt2 mail_tt2.new/automatic_bounce_management.tt2
--- mail_tt2/automatic_bounce_management.tt2	2016-08-31 16:53:41.533193643 +0200
+++ mail_tt2.new/automatic_bounce_management.tt2	2016-09-01 15:03:03.554362002 +0200
@@ -15,4 +15,4 @@
 [% user %]
 [% END %]
 
-[%|loc%]Managing bouncing subscribers:[%END%] [% conf.wwsympa_url %]/reviewbouncing/[% list.name %]
+[%|loc%]For further information, please contact the forge administrator.[%END%]
diff -Naur mail_tt2/command_report.tt2 mail_tt2.new/command_report.tt2
--- mail_tt2/command_report.tt2	2016-08-31 16:53:41.545193678 +0200
+++ mail_tt2.new/command_report.tt2	2016-09-01 15:03:03.554362002 +0200
@@ -189,4 +189,4 @@
 
 -----------==========================================-----------
 
-[%|loc(conf.wwsympa_url)%]For further information, check the mailing list web site %1[%END%]
+[%|loc%]For further information, please contact the forge administrator.[%END%]
diff -Naur mail_tt2/d_install_shared.tt2 mail_tt2.new/d_install_shared.tt2
--- mail_tt2/d_install_shared.tt2	2016-08-31 16:53:41.572193756 +0200
+++ mail_tt2.new/d_install_shared.tt2	2016-09-01 15:03:03.554362002 +0200
@@ -4,4 +4,4 @@
 [%|loc(filename,list.name,list.host,installed_by)%]Your document %1 for list %2@%3
 has been installed by %4 list editor.[%END%]
 
-[%|loc%]The list document repository:[%END%] [% conf.wwsympa_url %]/d_read/[% list.name %]
+[%|loc%]To consult the list document repository, please contact the forge administrator.[%END%]
diff -Naur mail_tt2/expire_warning1.tt2 mail_tt2.new/expire_warning1.tt2
--- mail_tt2/expire_warning1.tt2	2016-08-31 16:53:41.573193759 +0200
+++ mail_tt2.new/expire_warning1.tt2	2016-09-01 15:03:03.554362002 +0200
@@ -5,7 +5,7 @@
   [% conf.email %]@[% conf.host %]
 [%~ END %]
 
-[%|loc%]Everything about this list: [%END%][% conf.wwsympa_url %]/info/[% list.name %]
+[%|loc(list.name)%]To get information about this %1 list, please contact the forge administrator.[%END%]
 [%|loc%]Unsubscription:[%END%]
 [% | mailtourl({subject => "sig ${list.name} ${user.email}"}) ~%]
   [% conf.email %]@[% conf.host %]
diff -Naur mail_tt2/expire_warning2.tt2 mail_tt2.new/expire_warning2.tt2
--- mail_tt2/expire_warning2.tt2	2016-08-31 16:53:41.574193762 +0200
+++ mail_tt2.new/expire_warning2.tt2	2016-09-01 15:03:03.554362002 +0200
@@ -6,7 +6,7 @@
   [% conf.email %]@[% conf.host %]
 [%~ END %]
 
-[%|loc%]Everything about this list: [%END%][% conf.wwsympa_url %]/info/[% list.name %]
+[%|loc(list.name)%]To get information about this %1 list, please contact the forge administrator.[%END%]
 [%|loc%]Unsubscription:[%END%]
 [% | mailtourl({subject => "sig ${list.name} ${user.email}"}) ~%]
   [% conf.email %]@[% conf.host %]
diff -Naur mail_tt2/list_created.tt2 mail_tt2.new/list_created.tt2
--- mail_tt2/list_created.tt2	2016-08-31 16:53:41.586193796 +0200
+++ mail_tt2.new/list_created.tt2	2016-09-01 15:03:03.554362002 +0200
@@ -1,9 +1,3 @@
 Subject: [% FILTER qencode %][%|loc(list.name)%]%1 mailing list creation[%END%][%END%]
 
-[%|loc(list.name,list.host)%]%1@%2 mailing list has been activated by listmaster.[%END%]
-
-[%|loc%]homepage[%END%]
-[% conf.wwsympa_url %]/info/[% list.name %]
-
-[%|loc(list.name)%]%1 admin page[%END%]
-[% conf.wwsympa_url %]/admin/[% list.name %]
+[%|loc(list.name,list.host)%]%1@%2 mailing list has been activated by listmaster.[%END%]
\ No newline at end of file
diff -Naur mail_tt2/listeditor_notification.tt2 mail_tt2.new/listeditor_notification.tt2
--- mail_tt2/listeditor_notification.tt2	2016-08-31 16:53:41.600193837 +0200
+++ mail_tt2.new/listeditor_notification.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -3,20 +3,20 @@
 Subject: [% FILTER qencode %][%|loc(list.name)%]Shared document to be approved for %1[%END%][%END%]
 
 [% IF many_files %]
-[%|loc(list.name,filename,who,conf.wwsympa_url)%]There are new shared documents in list %1: 
+[[%|loc(list.name,filename,who)%]There are new shared documents in list %1: 
 	%2
 
 	from %3
 
-To moderate these document: 
-%4/modindex/%1[% END %]
+
+[% END %]
 
 [% ELSE %]
-[%|loc(list.name,filename,who,conf.wwsympa_url)%]There is a new shared document in list %1: 
+[[%|loc(list.name,filename,who)%]There is a new shared document in list %1: 
 	%2 from %3
 
-To moderate this document: 
-%4/modindex/%1
+
+
  [% END %][% END %]
 
 [% ELSE -%]
@@ -24,3 +24,5 @@
 
 [% param0 %]
 [% END %]
+
+[%|loc%]For further information, please contact the forge administrator.[%END%]
\ No newline at end of file
diff -Naur mail_tt2/listmaster_notification.tt2 mail_tt2.new/listmaster_notification.tt2
--- mail_tt2/listmaster_notification.tt2	2016-08-31 16:53:41.610193865 +0200
+++ mail_tt2.new/listmaster_notification.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -6,10 +6,10 @@
 
 [% list.name %]@[% list.host %]
 [% list.subject %]
-[% conf.wwsympa_url %]/info/[% list.name %]
 
-[%|loc%]To activate/delete this mailing list:[%END%]
-[% conf.wwsympa_url %]/ticket/[%one_time_ticket%]
+
+
+
 
 [%- ELSIF type == 'list_created' -%]
 [% PROCESS 'list_created.tt2' -%]
@@ -28,8 +28,8 @@
 [%|loc(email,old_listname,new_listname)%]%1 has renamed list "%2" to "%3"[%END%]
 [% END %]
 
-[%|loc%]To activate/delete this mailing list:[%END%]
-[% conf.wwsympa_url %]/ticket/[%one_time_ticket%]
+
+
 
 [% ELSIF type == 'no_db' -%]
 Subject: [% FILTER qencode %][%|loc%]No DataBase[%END%][%END%]
@@ -338,7 +338,7 @@
 
 [%|loc%]See logs for more details.[%END%]
 
-[%|loc%]Check the bounces in this list:[% END %] [% conf.wwsympa_url %]/reviewbouncing/[% listname %]
+
 
 [% ELSIF type == 'automatic_list_creation_failed' -%]
 Subject: [% FILTER qencode %][%|loc%]Listmaster: internal server error [%END%][%END%]
@@ -439,3 +439,5 @@
 
 [% param0 %]
 [% END %]
+
+[%|loc%]For further information, please contact the forge administrator.[%END%] 
diff -Naur mail_tt2/listowner_notification.tt2 mail_tt2.new/listowner_notification.tt2
--- mail_tt2/listowner_notification.tt2	2016-08-31 16:53:41.627193914 +0200
+++ mail_tt2.new/listowner_notification.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -29,10 +29,10 @@
 [% ELSIF type == 'warn-signoff' -%]
 Subject: [% FILTER qencode %][%|loc(type,list.name,who,gecos)%]FYI: %1 List "%2" from %3 %4[%END%][%END%]
 
-[%|loc(who,gecos,list.name)%]WARNING: %1 %2 failed to unsubscribe from %3 because his address was not found in the list.
-You may help this person looking for similar email in subscriber list using the following link :[%END%]
+[%|loc(who,gecos,list.name)%]WARNING: %1 %2 failed to unsubscribe from %3 because his address was not found in the list.[%END%]
+
+
 
-[% conf.wwsympa_url %]/ticket/[% one_time_ticket %]
  
 [% ELSIF type == 'erase_customizing' -%]
 Subject: [% FILTER qencode %][%|loc(list.name)%]List "%1" customizations have been removed[%END%][%END%]
@@ -49,10 +49,10 @@
 
 [%|loc(list.name)-%]Dear owner of list %1,
  
-A user asked to be added as a subscriber to your list. In order to list and manage
-pending subscription requests you should click on the following link:[%END%]
+A user asked to be added as a subscriber to your list. 
+[%END%]
+
 
-[% conf.wwsympa_url %]/ticket/[% one_time_ticket %]
 
 [%|loc(replyto)%]If prefered, you can accept this subscription request sending an email to %1 with the following subject:[%END%]
 AUTH [%keyauth%] ADD [%list.name%] [%who%] [%gecos%]
@@ -95,7 +95,7 @@
 
 You should delete bouncing subscribers: 
 [%END%]
-[% conf.wwsympa_url %]/reviewbouncing/[% list.name %]
+[%|loc%]Please contact the forge administrator.[%END%]
 
 
 [% ELSIF type == 'notice' -%]
@@ -147,3 +147,5 @@
 
 [% param0 %]
 [% END %]
+
+[%|loc%]For further information, please contact the forge administrator.[%END%]
\ No newline at end of file
diff -Naur mail_tt2/moderate.tt2 mail_tt2.new/moderate.tt2
--- mail_tt2/moderate.tt2	2016-08-31 16:53:41.652193986 +0200
+++ mail_tt2.new/moderate.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -14,7 +14,7 @@
 
 [% IF method == 'md5' -%]
 [%|loc(mod_spool_size)%]They are %1 messages awaiting moderation.[%END%] 
-[%|loc%]To browse the awaiting messages, you should click on the following link:[%END%] <[% conf.wwsympa_url %]/ticket/[% one_time_ticket %]>
+
 
 [% IF request_topic -%][%|loc()%]This mailing list is configured to require topics; that's probably why this message went through the moderation process.[%END%]
 
@@ -38,7 +38,7 @@
 [%|loc(conf.email,conf.host)%]Or send a message to %1@%2 with the following subject:[%END%]
 REJECT [% list.name %] [% modkey %]
 [%END%]
-[%|loc(conf.wwsympa_url)%]The messages moderating documentation:[%END%] <[% conf.wwsympa_url %]/help/admin#moderate>
+[%|loc%]For further information, please contact the forge administrator.[%END%]
 [%- END%]
 [%- END%]
 [% IF spam_status != 'spam' -%]
diff -Naur mail_tt2/modindex.tt2 mail_tt2.new/modindex.tt2
--- mail_tt2/modindex.tt2	2016-08-31 16:53:41.666194026 +0200
+++ mail_tt2.new/modindex.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -7,7 +7,7 @@
 
 [%|loc(total,list.name,list.host)%]There are %1 messages to be moderated for list %2@%3[%END%]
 
-[%|loc%]The messages moderating documentation:[%END%] <[% conf.wwsympa_url %]/help/admin#moderate>
+[%|loc%]For further information, please contact the forge administrator.[%END%]
 
 [% IF spool %]
 --[% boundary1 %]
diff -Naur mail_tt2/reject.tt2 mail_tt2.new/reject.tt2
--- mail_tt2/reject.tt2	2016-08-31 16:53:41.685194081 +0200
+++ mail_tt2.new/reject.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -7,5 +7,5 @@
 [%|loc(subject)%](Subject of your mail: %1)[%END%]
 
 
-[%|loc(list.name)%]Check %1 list usage:[%END%]
-[% conf.wwsympa_url %]/info/[% list.name %]
+[%|loc(list.name)%]To get information about this %1 list, please contact the forge administrator.[%END%]
+
diff -Naur mail_tt2/remind.tt2 mail_tt2.new/remind.tt2
--- mail_tt2/remind.tt2	2016-08-31 16:53:41.710194152 +0200
+++ mail_tt2.new/remind.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -3,7 +3,7 @@
 
 [%|loc(list.name,list.host,user.email,user.password)%]Your are subscriber of list %1@%2 with  email %3[%END%]
 
-[%|loc%]Everything about this list: [%END%][% conf.wwsympa_url %]/info/[% list.name %]
+[%|loc(list.name)%]To get information about this %1 list, please contact the forge administrator.[%END%]
 [% IF subscriber.included != 1 %]
 [%|loc%]Unsubscription:[%END%]
 [% | mailtourl({subject => "sig ${list.name} ${user.email}"}) ~%]
diff -Naur mail_tt2/send_auth.tt2 mail_tt2.new/send_auth.tt2
--- mail_tt2/send_auth.tt2	2016-08-31 16:53:41.712194158 +0200
+++ mail_tt2.new/send_auth.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -7,9 +7,9 @@
 --[% boundary %]
 Content-Disposition: inline
 
-[% IF request_topic -%][%|loc(list.name,conf.wwsympa_url,authkey)%]List %1 allows you to tag your messages, using a set of defined topics. To tag the attached message go to the following page:[%END%]
+[% IF request_topic -%][%|loc(list.name)%]List %1 allows you to tag your messages, using a set of defined topics.[%END%]
 
-[%conf.wwsympa_url%]/request_topic/[%list.name%]/[%authkey%]
+[%|loc(list.name)%]To get information about this %1 list, please contact the forge administrator.[%END%]
 [%- ELSE -%][%|loc(list.name)%]To distribute the attached message in list %1, follow the link below:[%END%]
 
 [% | mailtourl({subject => "CONFIRM ${authkey}"}) ~%]
diff -Naur mail_tt2/summary.tt2 mail_tt2.new/summary.tt2
--- mail_tt2/summary.tt2	2016-08-31 16:53:41.737194231 +0200
+++ mail_tt2.new/summary.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -6,7 +6,9 @@
 
 [% FOREACH m = msg_list -%]
 [%|wrap(0,2)%]* [% m.id %] - [% m.subject %] - [% m.from %][%END#wrap%]
-[% IF conf.wwsympa_url -%]
-  [% conf.wwsympa_url %]/arcsearch_id/[% list.name %]/[% m.month %]/[% m.message_id %]
-[% END %]
+
+  
+
 [% END -%]
+
+[%|loc(list.name)%]To get information about this %1 list, please contact the forge administrator.[%END%]
\ No newline at end of file
diff -Naur mail_tt2/user_notification.tt2 mail_tt2.new/user_notification.tt2
--- mail_tt2/user_notification.tt2	2016-08-31 16:53:41.747194260 +0200
+++ mail_tt2.new/user_notification.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -18,8 +18,8 @@
    [% t %]
 [% END -%]
 
-[%|loc%]To update your topics subscription, go to the following page:[%END%]
-[% conf.wwsympa_url %]/suboptions/[% list.name %]
+[%|loc%]To update your topics subscription, please contact the forge administrator.[%END%]
+
 
 
 [% ELSIF type == 'added_as_listadmin' -%]
@@ -31,15 +31,15 @@
 [%|loc(delegator,list.name,list.host)%]You have been delegated the responsability of list moderator by %1 for list %2@%3.[%END%]
 [% END %]
 
-[%|loc%]The list homepage:[%END%] [% conf.wwsympa_url %]/info/[% list.name %]
-[%|loc%]Owner and moderator guide:[%END%] [% conf.wwsympa_url %]/help/admin
+
+
 
 
 [% ELSIF type == 'ticket_to_send' -%]
 Subject: [% FILTER qencode %][%|loc%]Changing your subscriber email address[%END%][%END%]
 
-[%|loc%]You have requested to change your subscriber email address. To confirm your new email address with the SYMPA server, please click the link below:[%END%] 
-  [% conf.wwsympa_url %]/ticket/[% one_time_ticket %]
+[%|loc%]You have requested to change your subscriber email address. To confirm your new email address with the SYMPA server, please contact the forge administrator.[%END%] 
+
 
 
 [% ELSIF type == 'ticket_to_signoff' -%]
@@ -49,9 +49,9 @@
 This query was issued from the IP number '%2'.
 To confirm and get removed from the list, please click the link below: [%END%] 
 [% ELSE %]
-[%|loc(list.name)%]You have requested to be removed from list %1. To confirm and get removed from the list, please click the link below: [%END%] 
+[%|loc(list.name,conf.listmaster_email,conf.host)%]You have requested to be removed from list %1. To confirm and get removed from the list, please contact the forge administrator %2@%3.[%END%]
 [% END %]
-  [% conf.wwsympa_url %]/ticket/[% one_time_ticket %]
+
 
 [% ELSIF type == 'ticket_to_family_signoff' -%]
 Subject: [% FILTER qencode %][%|loc(family)%]Unsubscribing from family %1[%END%][%END%]
@@ -60,9 +60,9 @@
 This query was issued from the IP number '%2'.
 To confirm and get removed from all the lists of this family, please click the link below: [%END%] 
 [% ELSE %]
-[%|loc(family)%]You have requested to be removed from family %1. To confirm and get removed from all the lists of this family, please click the link below: [%END%] 
+[%|loc(family)%]You have requested to be removed from family %1. To confirm and get removed from all the lists of this family, please contact the forge administrator.[%END%]
 [% END %]
-  [% conf.wwsympa_url %]/ticket/[% one_time_ticket %]
+
 
 
 [% ELSIF type == 'hundred_percent_error' -%]
diff -Naur mail_tt2/welcome.tt2 mail_tt2.new/welcome.tt2
--- mail_tt2/welcome.tt2	2016-08-31 16:53:41.747194260 +0200
+++ mail_tt2.new/welcome.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -9,6 +9,6 @@
 [% CATCH %]
 [% END %]
 
-[%|loc%]The list homepage:[%END%] [% conf.wwsympa_url %]/info/[% list.name %]
-[%|loc%]General informations about mailing lists:[%END%] [% conf.wwsympa_url %]/help/introduction
 
+
+[%|loc(list.name)%]To get information about this %1 list, please contact the forge administrator.[%END%]
\ No newline at end of file
diff -Naur mail_tt2/x509-user-cert-missing.tt2 mail_tt2.new/x509-user-cert-missing.tt2
--- mail_tt2/x509-user-cert-missing.tt2	2016-08-31 16:53:41.748194263 +0200
+++ mail_tt2.new/x509-user-cert-missing.tt2	2016-09-01 15:03:03.555362005 +0200
@@ -9,5 +9,5 @@
 [%|loc(conf.email,conf.host)%]Please, in order to receive future encrypted messages send a signed message
 to  %1@%2 .[%END%]
 
-[%|loc%]Information about this list:[%END%]
-[% conf.wwsympa_url %]/info/[% list.name %]
+[%|loc(list.name)%]To get information about this %1 list, please contact the forge administrator.[%END%]
+
